woocommerce/woocommerce-gutenberg-products-block
# Sync order data with cart data when cart is updated from any route

Syncs cart to draft order (if one exists) on all POST PUT DELETE requests so draft orders are always current, even on the cart page.

Fixes #5366

### Testing

How to test the changes in this Pull Request:

Ensure you have a store with products and at least 2 shipping methods.

1. Add an item to the cart.
2. Visit the checkout page.
3. In another tab, open admin and view the draft order. Note the shipping method.
4. Leave checkout and go to the cart page. Change the shipping method.
5. In your admin tab, refresh the order. Confirm the shipping method + totals all updated.
6. Smoke test checkout/cart to ensure totals, taxes, items, and fee totals all make sense and change when the cart changes.

### Performance Impact

Does write to orders more often so API requests may slow down due to this.

### Changelog

> Sync draft orders whenever cart data changes.
